sdk/python/kfp/dsl/io_types.py

```diff
@@ -53,13 +55,22 @@ def path(self):
   def path(self, path):
     self._set_path(path)
 
-  def _get_path(self) -> str:
+  def _get_path(self) -> Optional[str]:
     if self.uri.startswith('gs://'):
       return _GCS_LOCAL_MOUNT_PREFIX + self.uri[len('gs://'):]
+    elif self.uri.startswith('minio://'):
+      return _MINIO_LOCAL_MOUNT_PREFIX + self.uri[len('minio://'):]
+    elif self.uri.startswith('s3://'):
+      return _S3_LOCAL_MOUNT_PREFIX + self.uri[len('s3://'):]
+    return None
 
   def _set_path(self, path):
```

Comment:
nit: not related to your change, but I now feel like this method name is a bit confusing, we can setting self.uri but not path. Maybe we could rename this private method to `_set_uri` while keep the `path` property name unchanged. So it would be 
```
@path.setter
def path(self, path):
  self._set_uri(path)
```
You don't have to make the change though. This isn't related to your change. We may clean things up later.

Reply:
Agree with these points, I felt it was confusing too.